src/store/agreement/agreementSlice.ts

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable functional/immutable-data */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Agreement } from '../../api/generated';
+import Api from '../../api/index';
+import { setCookie } from '../../utils/cookie';
+
+export const createAgreement = createAsyncThunk('agreement/createStatus', async () => {
+	const response = await Api.Agreement.createAgreement();
+	if (response.data) {
+		setCookie(response.data.id);
+	}
+	return response.data;
+});
```

Comment:
This can be refactored using `TaskEither` and `Option` for `response.data`. Something like:
```suggestion
export const createAgreement = createAsyncThunk('agreement/createStatus', async () => {
	await tryCatch(() => Api.Agreement.createAgreement(), toError)
	.map(response => 
		fromNullable(response.data).foldL(() => response.data, _ => {
			setCookie(_.id);
			return _;
		})
	)
	.fold(
		() => void 0,
		identity
	).run();
});
```